cli/source/app.tsx

```diff
@@ -264,12 +299,37 @@ const CliUi = () => {
     return () => {
       globalThis.process.off('exit', handleExit);
     };
-  }, [dispatch, exit, handleExit, runCommandOpts, retryToggle]);
+  }, [dispatch, exit, handleExit, runCommandOpts, retryToggle, appState]);
+
+  useEffect(() => {
+    preCheck.callChecks();
+  }, []);
 
   const logsStreamNodes = useMemo(
     () => logsStream.map((l) => transformLogToNode(l)),
     [logsStream]
   );
+
+  const PreCheckDisplayElement = ({
+    value,
+    property
+  }: {
+    value: PreCheckStates;
+    property: string;
```

Comment:
@sidtohan can you declare type for `property`

Reply:
Sure, I'll add an enum for this.